builder/metadata.go

```diff
@@ -149,6 +151,48 @@ func (b *Builder) getUpstreamFormatRange(version string) (format string, first,
 	return format, first, latest, err
 }
 
+// ModifyBundles goes through the bundle directory and performs an action when it finds
+// a Deprecated bundle
+func (b *Builder) ModifyBundles(action func(string) error) error {
+	path := b.Config.Mixer.LocalBundleDir
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	var scanner *bufio.Scanner
+	for _, file := range files {
+		fileToScan := filepath.Join(path, file.Name())
+		f, err := os.Open(fileToScan)
+		if err != nil {
+			return err
+		}
+
+		// Scan the files and find which bundle definitions are marked deprecated
+		scanner = bufio.NewScanner(f)
+		var str string
+		re := regexp.MustCompile("#\\s\\[STATUS\\]:\\s*Deprecated.*")
+		for scanner.Scan() {
+			str = scanner.Text()
+			// Don't scan past header, stop once we have no more # comments
+			if str[0] == '#' {
+				if index := re.FindStringIndex(str); index != nil {
```

Comment:
Instead of `FindStringIndex` here (which requires you to check `index` against `nil`) just look for a `re.Match`

```go
if re.Match(str) {
    // code
}
```

https://golang.org/pkg/regexp/#Regexp.Match

Reply:
ahhh thx